Automattic/wp-calypso
# Plans: Implement plan and product upsells for Jetpack backup product

This PR implements upsell buttons for the Jetpack backup single products. This functionality supports a couple of use cases, and to function it requires a map of product features and their higher tier upsells:

* Suggest a higher-tier product when we have already purchased a lower-tier product.
* Suggest a higher-tier plan with the higher tier product when we have already purchased a plan that includes the lower tier of the product.

#### Changes proposed in this Pull Request

* Plans: Implement plan and product upsells for Jetpack backup product

#### Preview

Higher-tier product upsell:

![](https://cldup.com/nHuZRPRbUS.png)

Upsell with a plan that offers a higher-tier product:

![](https://cldup.com/HzyxQfd3_5.png)

#### Testing instructions

* Checkout this branch.
* For a Jetpack site, verify you have a Jetpack backup daily yearly product.
* Go to http://calypso.localhost:3000/plans/:site where `:site` is the Jetpack site slug.
* Verify you get the "realtime yearly" suggested as an upsell purchase.
* For another Jetpack site, verify you have a Jetpack personal plan.
* Go to http://calypso.localhost:3000/plans/:site where `:site` is the Jetpack site slug.
* Verify you get the Jetpack professional plan suggested as an upsell purchase.
* Verify the devdocs example http://calypso.localhost:3000/devdocs/blocks/product-selector still works well.
